spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/dao/JdbcTaskExecutionDao.java

```diff
@@ -0,0 +1,162 @@
+		Assert.notNull(jdbcTemplate);
+		this.jdbcTemplate = jdbcTemplate;
+	}
+
+	@Override
+	public void saveTaskExecution(TaskExecution taskExecution) {
+		validateTaskExecution(taskExecution);
+
+		Object[] parameters = new Object[]{ taskExecution.getExecutionId(),
+				taskExecution.getStartTime(), taskExecution.getEndTime(),
+				taskExecution.getTaskName(), taskExecution.getExitCode(),
+				taskExecution.getExitMessage(), new Date(),
+				taskExecution.getStatusCode() };
+		int addCount = getJdbcTemplate().update(
+				getQuery(SAVE_TASK_EXECUTION),
+				parameters,
+				new int[]{ Types.VARCHAR, Types.TIMESTAMP, Types.TIMESTAMP, Types.VARCHAR,
+						Types.INTEGER, Types.VARCHAR, Types.TIMESTAMP, Types.VARCHAR });
+		insertJobParameters(taskExecution.getExecutionId(), taskExecution.getParameters());
+	}
+
+	@Override
+	public void updateTaskExecution(TaskExecution taskExecution) {
+
+		validateTaskExecution(taskExecution);
+
+		// Check if given TaskExecution's Id already exists, if none is found
+		// it is invalid and an exception should be thrown.
+		if (getJdbcTemplate().queryForObject(getQuery(CHECK_TASK_EXECUTION_EXISTS), Integer.class,
+				new Object[]{ taskExecution.getExecutionId() }) != 1) {
+			throw new IllegalStateException("Invalid TaskExecution, ID " + taskExecution.getExecutionId() + " not found.");
+		}
+
+		Object[] parameters = new Object[]{ taskExecution.getStartTime(), taskExecution.getEndTime(),
+				taskExecution.getTaskName(), taskExecution.getExitCode(),
+				taskExecution.getExitMessage(), new Date(), taskExecution.getStatusCode(),
+				taskExecution.getExecutionId() };
+		int count = getJdbcTemplate().update(
+				getQuery(UPDATE_TASK_EXECUTION),
+				parameters,
+				new int[]{ Types.TIMESTAMP, Types.TIMESTAMP, Types.VARCHAR, Types.INTEGER,
+						Types.VARCHAR, Types.TIMESTAMP, Types.VARCHAR, Types.VARCHAR });
+	}
+
+	/**
+	 * Public setter for the table prefix property. This will be prefixed to all
+	 * the table names before queries are executed. Defaults to
+	 * {@link #DEFAULT_TABLE_PREFIX}.
+	 *
+	 * @param tablePrefix the tablePrefix to set
+	 */
+	public void setTablePrefix(String tablePrefix) {
+		this.tablePrefix = tablePrefix;
+	}
+
+	public void setJdbcTemplate(JdbcOperations jdbcTemplate) {
+		this.jdbcTemplate = jdbcTemplate;
+	}
+
+	protected JdbcOperations getJdbcTemplate() {
+		return jdbcTemplate;
+	}
+
+	/**
+	 * Validate TaskExecution. At a minimum a startTime.
+	 *
+	 * @param taskExecution the taskExecution to be evaluagted.
+	 */
+	private void validateTaskExecution(TaskExecution taskExecution) {
+		Assert.notNull(taskExecution, "taskExecution should not be null");
+		Assert.notNull(taskExecution.getExecutionId(), "taskExecutionId should not be null");
+		Assert.notNull(taskExecution.getStartTime(), "TaskExecution start time cannot be null.");
+	}
+
+	private String getQuery(String base) {
+		return StringUtils.replace(base, "%PREFIX%", tablePrefix);
+	}
```

Comment:
We should think about a way to do this only once on creation instead of every time we request the query